libs/header/store/store.go

```diff
@@ -283,6 +283,10 @@ func (s *Store[H]) Has(ctx context.Context, hash header.Hash) (bool, error) {
 	return s.ds.Has(ctx, datastore.NewKey(hash.String()))
 }
 
+func (s *Store[H]) HasAt(height uint64) bool {
+	return height != uint64(0) && s.Height() >= height
+}
```

Comment:
Let's make ctx as a first param. 

Right now, we assume that the Store stores everything below `Height`. However, we will break this assumption once we will introduce pruning. Then we will do a disk look up which requires context and breaking the interface. Let's avoid breaking in it in future by introducing future proof form of it

Reply:
fine w me

Reply:
true. Good idea, thank you.